### ISSUE TITLE
New Resource: aws_lakeformation_resource_link

### ISSUE THREAD

**Issue by WafflesMcDuff** [hashicorp/terraform-provider-aws#17069]
Need the ability to create resource links through terraform for aws_lakeformation
![image](https://user-images.githubusercontent.com/24410823/104346623-a187ea00-54cd-11eb-973f-ab8936c7d601.png)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* aws_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by anGie44** [comment]
Hi @WafflesMcDuff, thank you for creating this issue! Making a first pass at the AWS LakeFormation User Guide/available SDK methods, I don't currently see support for this operation but we'll keep an eye out for this functionality 👍

**Comment by chrisbulgaria** [comment]
Hi, I think this is implemented on the underlying Glue level - here : https://docs.aws.amazon.com/glue/latest/webapi/API_CreateTable.html   (TargetDatabase) . But its indeed not available on the terraform glue resources, there is an issue for that already
https://github.com/hashicorp/terraform-provider-aws/issues/15296

**Comment by WafflesMcDuff** [comment]
Related to #15296

**Comment by amirbrown** [comment]
> Hi, I think this is implemented on the underlying Glue level - here : https://docs.aws.amazon.com/glue/latest/webapi/API_CreateTable.html (TargetDatabase) . But its indeed not available on the terraform glue resources, there is an issue for that already
> #15296

Yeah, lf is confusing because the console ui fronts various parts of glue functionality without being clear what it is doing. 
If you look at the lf docs about the resource links you can see that it is all glue apis and permissions.
https://docs.aws.amazon.com/lake-formation/latest/dg/resource-links-glue-apis.html 

There is no reference to a resource link in the lf api docs or lf cloudformation docs.

**Comment by rito-sixt** [comment]
Hi. Any updates on this?

**Comment by Mralbert93** [comment]
You should be able to create resource links in Lake Formation by using the `aws_glue_catalog_database` resource and specifying a `target_database` block.
